### DISCUSSION TITLE
Create Automated Tests for Repo

### DISCUSSION THREAD

**Discussion by Zach-Attach** [buildingamind/NewbornEmbodiedTuringTest#156] [Ideas]
We should set up github actions to automate testing when merging code into dev or main. I have opened[ a branch ](https://github.com/buildingamind/NewbornEmbodiedTuringTest/tree/tests) to create these tests, but, currently they fail as a result of needing an nvidia gpu. 

Proposed solution:
Use one of our servers to run these tests in a docker container using [CML](https://cml.dev/doc). This requires sudo access on the server.

**Comment by desaibhargav** [comment]
Thanks for creating this issue and bringing this to my attention. By design, we want to be able to run this code on CPU as well, so it shouldn't have hard dependencies on NVIDIA. I will create a branch to fix this before our next release. 

That being said, I'm not entirely convinced yet that we need to run tests on a server with GPU at this point. Because we're using `stable-baselines3` / `torch` under the hood, I would be fairly confident that if a GPU device is available on the user's compute setup, the code should run as intended. The downside is that some areas of the codebase would not be covered by tests, which I would be okay with at this point. I would prefer to have a CPU runner for tests for now, reducing any complications with setting up tests on a GPU runner. 

I'm open to discussion on this though, but I'd rater move this to discussions than keep it here.

**Comment by Zach-Attach** [comment]
I was just thinking of enforcing a simple test to make sure that if you give it an executable and the right data it will work. Basically, we could just check that all of the files are there after it runs and that it doesnt spit out an error. Pretty simple and would improve the PR process so that we don't break things

**Comment by Zach-Attach** [comment]
I have found that the CPU mode does not work for our needs as we will need the images from Unity which would be incompatible with the CPU mode, so we will need to run these on one of our own servers. I am thinking we could try to run CML on dog.

**Comment by desaibhargav** [comment]
I moved this here because I'm interested in taking this up. But I wanted to discuss a strategy for how we want to do this first before tracking it as an issue.
